src/libcore/lib.rs

```diff
@@ -60,8 +60,8 @@
 
 #![warn(deprecated_in_future)]
 #![warn(missing_docs)]
```

Comment:
Btw @alexcrichton is there any reason liballoc does not have `warn(missing_docs)`? Should I add it there as well to make things more consistent?

The other inconsistency I noticed is that libcore does not have `deny(rust_2018_idioms)`. But that seems people just didn't get there yet.

Reply:
It is being worked on, https://github.com/rust-lang/rust/pull/58702.

Reply:
@RalfJung when those annotations were added it wasn't needed, now that it's a stable crate we probably need it

Reply:
Okay, I added the attribute. That uncovered a few missing docs, and indeed for items that got reexported in `std`! So I also added docs.